### PR TITLE
Add keyboard shortcut for triggering abort action; #1191

### LINKED ISSUE #1191: Add keyboard shortcut for triggering abort action
It is sometimes annoying to click when results are scrolling fast. This should probably be accompanied with a tooltip (like on the judge status page) informing users of the keyboard shortcut.

### PR DESCRIPTION
**Pull request by Ninjaclasher**
Fixes #1191.

### CODE REVIEW

File: templates/submission/status.html [hunk 1]
```diff
@@ -52,6 +53,17 @@
         });
         </script>
     {% endif %}
+    {% if not submission.is_graded %}
+        <script type="text/javascript">
+            $(function () {
+                $(document).keydown(function(ev) {
```
Comment (TheAvidDev) [inline templates/submission/status.html:59]:
```suggestion
                $(document).keydown(function(e) {
```
We use `e` for events everywhere else, unless there's another reason?

File: templates/submission/status.html [hunk 2]
```diff
@@ -52,6 +53,17 @@
...
+                    if ((ev.metaKey || ev.ctrlKey) && ev.which == 13) {
```
Comment (TheAvidDev) [inline templates/submission/status.html:60]:
```suggestion
                    if ((e.metaKey || e.ctrlKey) && e.which == 13) {
```
Same as above.

File: templates/submission/status.html (hunk 1 above)
Comment (Ninjaclasher) [reply templates/submission/status.html:59]: This JS was copied from somewhere else in this repository, but I can't seem to find it anymore. Will change to `e`.

**Comment by TheAvidDev** [comment]
Actually, it seems like the better way to check keys is [`KeyboardEvent.code`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code), although it's [not supported](https://caniuse.com/#search=KeyboardEvent.code) by IE or mobile browsers. Not sure if we should switch to it.

**Comment by Ninjaclasher** [comment]
This is how the DMOJ has always handled keyboard shortcuts.

**Comment by TheAvidDev** [comment]
> This is how the DMOJ has always handled keyboard shortcuts.

Sounds alright then, but it _is_ [officially deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode) so we should eventually switched.

File: templates/submission/status.html [hunk 3]
```diff
@@ -2,7 +2,8 @@
 {% block js_media %}
     <script type="text/javascript" src="{{ static('event.js') }}"></script>
     {% if not submission.is_graded and last_msg %}
-        <script type="text/javascript">$(function () {
+        <script type="text/javascript">
+        $(function () {
```
Comment (Xyene) [inline templates/submission/status.html:6]: If we're fixing the spacing here, might as well fix the indentation too.

File: templates/submission/status.html [hunk 4]
```diff
@@ -52,6 +53,15 @@
         });
         </script>
     {% endif %}
+    <script type="text/javascript">
+        $(function () {
+            $(document).keydown(function(e) {
+                if ((e.metaKey || e.ctrlKey) && e.which == 13) {
```
Comment (Xyene) [inline templates/submission/status.html:59]: Should have a comment for those not fluent in keycode numbers :)

**Comment by Ninjaclasher** [comment]
> Otherwise looks great! Perhaps we can get a screenshot of what the form title looks like?

@TheAvidDev 
image

(I can't get the cursor to show up in the screenshot.)

File: templates/submission/status.html (hunk 3 above)
Comment (Ninjaclasher) [reply templates/submission/status.html]: Done

File: templates/submission/status.html (hunk 4 above)
Comment (Ninjaclasher) [reply templates/submission/status.html]: Done

**Pull request merged**